.github/workflows/prettier-format.yml

```diff
@@ -1,9 +1,54 @@
-name: Prettier Format
+name: Reformat Codebase with Prettier
+
 on:
-  workflow_dispatch:
+  workflow_dispatch:  # Allows you to trigger this workflow manually
+permissions:
+  contents: write
+
 jobs:
-  build:
+  reformat:
```

Comment:
this is excessive - eslint check will fail if there are changes coming from prettier. 
Also it is kinda scary to commit automatically generated changes without review - in theory it is safe but sometimes may be undesirable - in rare cases you might want to disable prettier for a code block to have better formatting through indentation by example

Reply:
this is not the intention. I should clarify the plan.

the sole purpose of this job is to provide a verifiable commit that reformats the whole codebase. It's easy to verify by reviewers using e.g. `git patch-id` .

the idea is to use merge this commit into this (or a separate PR), and do the rest of minor fixed manually on top

Reply:
after it is merged `eslint` target will be failing for every commit having unformatted code. It may be easier to wait until there is no active work and launch prettier and commit result once for whole codebase

Reply:
we might disable eslint rule for now

Reply:
>  It may be easier to wait until there is no active work and launch prettier and commit result once for whole codebase

Already done, this branch is rebased on the latest dev.